### DISCUSSION TITLE
How to use `error` and `loading` in parallel routes in "app" dir ?

### DISCUSSION THREAD

**Discussion by tresorama** [vercel/next.js#49588] [App Router]
Hi, i'm trying the "app" directory for first time following the official docs.

While testing "parallel routes" I wanted to test the error component.
Looking at this image in the Docs, it should be possible to render different `error` component (and `loading`) for different parallel routes.
![image](https://nextjs.org/_next/image?url=%2Fdocs%2Fdark%2Fparallel-routes-cinematic-universe.png&w=3840&q=75)

I defined an `error.tsx` file inside each `@folder` of a segment.
Then I triggered a forced error (inside a `fetch`) , I expect to see `segment/@folder/error.tsx` rendered, but instead I see `segment/error.tsx` rendered.

Same thing with `loading.tsx`, expecting `segment/@folder/loading.tsx` to be rendered and instead got  `segment/loading.tsx`.

> **Code demo in [codesandbox][sandbox-link]**.
In the demo the segment with parallel routes is under `app/jsonplaceholders`.

----

This prevent me to render different `error` component (and `loading`) for different parallel routes, forcing me to have a shared `error` (and `loading`) for the entire segment (that acts as parent for `@folder` parallel routes)

---

What i want is not possible ? Am I doing something wrong ?

[sandbox-link]: https://codesandbox.io/p/github/tresorama/test-next-app-directory/draft/test-parallel-routes-error?selection=%5B%7B%22endColumn%22%3A49%2C%22endLineNumber%22%3A6%2C%22startColumn%22%3A49%2C%22startLineNumber%22%3A6%7D%5D&file=%2Fsrc%2Fapp%2Fpage.tsx&layout=%257B%2522activeFilepath%2522%253A%2522%252Fsrc%252Fapp%252Fjsonplaceholders%252Ferror.tsx%2522%252C%2522openFiles%2522%253A%255B%2522%252Fsrc%252Fapp%252Fjsonplaceholders%252F%2540todos%252Ferror.tsx%2522%252C%2522%252Fsrc%252Fapp%252Fjsonplaceholders%252F%2540posts%252Ferror.tsx%2522%252C%2522%252Fsrc%252Fapp%252Fjsonplaceholders%252Ferror.tsx%2522%255D%252C%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522gitSidebarPanel%2522%253A%2522COMMIT%2522%252C%2522fullScreenDevtools%2522%253Afalse%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522vertical%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522DEVTOOLS_PANELS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522panelType%2522%253A%2522TABS%2522%252C%2522id%2522%253A%2522clhcr6tup00093b6pos1h7901%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clhcr6tup00093b6pos1h7901%2522%253A%257B%2522id%2522%253A%2522clhcr6tup00093b6pos1h7901%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522id%2522%253A%2522clhcr7bgv007v3b6p22tlckru%2522%257D%252C%257B%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A3000%252C%2522id%2522%253A%2522clhcr7c5s00d43b6ptzocapon%2522%252C%2522path%2522%253A%2522%252Fjsonplaceholders%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clhcr7c5s00d43b6ptzocapon%2522%257D%257D%252C%2522showSidebar%2522%253Atrue%252C%2522showDevtools%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A13.371757750135941%252C%2522editorPanelSize%2522%253A39.46327178708321%252C%2522devtoolsPanelSize%2522%253A45.71766155278537%257D

**Comment by bagus2x** [comment]
Having the same problem. Did you find solution?

**Reply by tresorama** [reply]
Nope! 
Tried again now with `13.4.3`

**Comment by tresorama** [comment]
Related:
https://github.com/vercel/next.js/issues/49243 Issue
https://github.com/vercel/next.js/discussions/50107#discussioncomment-5965956 discussion

**Reply by lu-zen** [reply]
The workaround is putting inside a route group:
@news/(news)/page.tsx,error.tsx,loading.tsx

**Reply by tresorama** [reply]
Tested and yes, it works!
Better than nothing for now!  

[Live Demo on codesandbox](https://codesandbox.io/p/github/tresorama/test-next-app-directory/draft/test-parallel-routes-error--maybe-fix-routes-group?file=%2FREADME.md%3A1%2C1&layout=%257B%2522sidebarPanel%2522%253A%2522GIT%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522panelType%2522%253A%2522TABS%2522%252C%2522id%2522%253A%2522cli8upmns01453d6pruis6hyj%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522panelType%2522%253A%2522TABS%2522%252C%2522id%2522%253A%2522cli8upmns01473d6p0usaj0o8%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B54.98566841821021%252C45.01433158178979%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522cli8upmns01453d6pruis6hyj%2522%253A%257B%2522id%2522%253A%2522cli8upmns01453d6pruis6hyj%2522%252C%2522activeTabId%2522%253A%2522cli8uubwa02dy3d6palog6jzp%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cli8upmns01443d6pu0v8gqs5%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FREADME.md%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Fapp%252Fpage.tsx%2522%252C%2522id%2522%253A%2522cli8upzuj01bz3d6pz7mmzv2t%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Fapp%252Fjsonplaceholders-route-group%252F%2540posts%252F%28posts%29%252Ferror.tsx%2522%252C%2522id%2522%253A%2522cli8urk4k01kq3d6pxjfonepb%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Fapp%252Fjsonplaceholders-route-group%252F%2540todos%252F%28todos%29%252Ferror.tsx%2522%252C%2522id%2522%253A%2522cli8usfi101pp3d6pzzpce3we%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Fapp%252F_components%252FMainNav.tsx%2522%252C%2522id%2522%253A%2522cli8uubwa02dy3d6palog6jzp%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%257D%252C%2522cli8upmns01473d6p0usaj0o8%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cli8upmns01463d6p8rsxrscs%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A3000%252C%2522path%2522%253A%2522%252Fjsonplaceholders-route-group%2522%257D%255D%252C%2522id%2522%253A%2522cli8upmns01473d6p0usaj0o8%2522%252C%2522activeTabId%2522%253A%2522cli8upmns01463d6p8rsxrscs%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

**Comment by tresorama** [comment]
With 13.4.7 this bug is now fixed!!
[codesandbox demo](https://codesandbox.io/p/github/tresorama/test-next-app-directory/draft/test-parallel-routes-error-with13-4-7?file=%2Fpackage.json%3A19%2C14&selection=%5B%7B%22endColumn%22%3A49%2C%22endLineNumber%22%3A6%2C%22startColumn%22%3A49%2C%22startLineNumber%22%3A6%7D%5D&layout=%257B%2522sidebarPanel%2522%253A%2522GIT%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522panelType%2522%253A%2522TABS%2522%252C%2522id%2522%253A%2522clj7z93ow009m3b6nfoiv66p5%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522panelType%2522%253A%2522TABS%2522%252C%2522id%2522%253A%2522clj7z93ow009n3b6nsjf1co79%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B42.96849640018311%252C57.03150359981689%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clj7z93ow009m3b6nfoiv66p5%2522%253A%257B%2522id%2522%253A%2522clj7z93ow009m3b6nfoiv66p5%2522%252C%2522activeTabId%2522%253A%2522clj7zpq8c04jn3b6nylerrn77%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fpackage.json%2522%252C%2522id%2522%253A%2522clj7zb1vc01bs3b6n6y7dcn2w%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Fapp%252Fjsonplaceholders%252F%2540todos%252Ferror.tsx%2522%252C%2522id%2522%253A%2522clj7znsmp042u3b6n29fabxtw%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Fapp%252Fjsonplaceholders%252F%2540posts%252Ferror.tsx%2522%252C%2522id%2522%253A%2522clj7zojib049d3b6nv5mgvv4k%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Fapp%252Fjsonplaceholders%252F%2540todos%252Floading.tsx%2522%252C%2522id%2522%253A%2522clj7zpq8c04jn3b6nylerrn77%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%257D%252C%2522clj7z93ow009n3b6nsjf1co79%2522%253A%257B%2522id%2522%253A%2522clj7z93ow009n3b6nsjf1co79%2522%252C%2522activeTabId%2522%253A%2522clj7zhkx303le3b6nkww12vgp%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clj7zdtyl02zn3b6nm4b9mzhh%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522clj7zdymy001nfoj52cr61l41%2522%257D%252C%257B%2522id%2522%253A%2522clj7zhcmi03e23b6n3r2sw7wg%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%252C%257B%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A3000%252C%2522id%2522%253A%2522clj7zhkx303le3b6nkww12vgp%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252Fjsonplaceholders%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A13.371757750135941%257D)

Closing this...

**Reply by tresorama** [reply]
With 13.4.7 the bug is fixed, but on 13.5 the bug happens again.

**Comment by parirox** [comment]
but I also have the same issue in next 13.5.* , even though everything works properly in 13.4.7.

**Comment by NCSmit** [comment]
Hey devs, this routing method works correctly in 13.4.7 but it seems the bug has re-appeared in 13.5^.

Can we open this issue @tresorama ??

**Comment by tresorama** [comment]
> Hey devs, this routing method works correctly in 13.4.7 but it seems the bug has re-appeared in 13.5^.

Reopening this discussion because of this.

**Comment by tresorama** [comment]
Testing with `13.5.4`.
It's working as expected.
[codesandbox demo](https://codesandbox.io/p/github/tresorama/test-next-app-directory/draft/test-parallel-routes-error-with-13-5?layout=%257B%2522sidebarPanel%2522%253A%2522GIT%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clnhcfi9h00pj3b6lnmyfr3o0%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clnhcfi9h00pd3b6lsxduutbk%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clnhcfi9h00pg3b6lo67cmte3%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clnhcfi9h00pi3b6l1d9skltu%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clnhcfi9h00pd3b6lsxduutbk%2522%253A%257B%2522id%2522%253A%2522clnhcfi9h00pd3b6lsxduutbk%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clnhcfi9g00pc3b6lmo41542e%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FREADME.md%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fpackage.json%2522%252C%2522id%2522%253A%2522clnhcfpqs00u33b6l33ujttc9%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clnhcfpqs00u33b6l33ujttc9%2522%257D%252C%2522clnhcfi9h00pi3b6l1d9skltu%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clnhcfi9h00ph3b6l2pzu5k57%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A3000%252C%2522path%2522%253A%2522%252Fjsonplaceholders%2522%257D%255D%252C%2522id%2522%253A%2522clnhcfi9h00pi3b6l1d9skltu%2522%252C%2522activeTabId%2522%253A%2522clnhcfi9h00ph3b6l2pzu5k57%2522%257D%252C%2522clnhcfi9h00pg3b6lo67cmte3%2522%253A%257B%2522id%2522%253A%2522clnhcfi9h00pg3b6lo67cmte3%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clnhcfi9h00pf3b6lu2yo4v2p%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522clnhc4efq001ldqhjct2x5391%2522%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522id%2522%253A%2522clnhcjwuh01693b6ln300anxj%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clnhcjwuh01693b6ln300anxj%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

---

@NCSmit and @ivi3  
Can you confirm the bug is no present with 13.5.4 ??

**Reply by parirox** [reply]
First of all I appreciated for your respond, I checked your demo code by the latest version of next.js and update my projects  part by part as you did, and finally I figure out the reason of my issue is I have no page beside the layout in my route so I don't add children in the layout.
If you remove the page inside the "jsonplaceholders" route you will see the loading doesn't work as expected.

**Reply by tresorama** [reply]
Your problem was that you haven't declared the layout file `app/jsonplaceholders/layout.tsx` or `app/jsonplaceholders/page.tsx` or that you haven't `app/jsonplaceholders/@todos/page.tsx`?

I don't have time now but I suggest you try also a "npm run build" and "npm run start" and check if all is working as expected because, in other similar discussions, some folks reported that sometimes the error is not present in "dev" but it's present in "production".

**Reply by parirox** [reply]
Yes, I have no page.tsx alongside my layout.tsx in the same route, and because of that, the loading doesn't work properly.
I don't have any issue in production, but when I set up PWA configs in my project, the parallel routes still don't work!
Parallel routes have been troublesome for me.

**Comment by Abukodonosor** [comment]
Problem in next15@rc!

**Reply by tresorama** [reply]
Hi, @Abukodonosor
Tried updating the codesanbox to next15rc and see no problem
[codesanbox](https://codesandbox.io/p/github/tresorama/test-next-app-directory/draft/test-parallel-routes-error-with-15rc?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clnhcfi9h00pj3b6lnmyfr3o0%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clnhcfi9h00pd3b6lsxduutbk%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clnhcfi9h00pg3b6lo67cmte3%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clnhcfi9h00pi3b6l1d9skltu%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B60.29495514409041%252C39.70504485590959%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clnhcfi9h00pd3b6lsxduutbk%2522%253A%257B%2522id%2522%253A%2522clnhcfi9h00pd3b6lsxduutbk%2522%252C%2522activeTabId%2522%253A%2522clyudvx8x00023b6k5b6aafqd%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clnhcfi9g00pc3b6lmo41542e%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FREADME.md%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fpackage.json%2522%252C%2522id%2522%253A%2522clnhcfpqs00u33b6l33ujttc9%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A2%252C%2522startColumn%2522%253A22%252C%2522endLineNumber%2522%253A2%252C%2522endColumn%2522%253A22%257D%255D%257D%252C%257B%2522id%2522%253A%2522clyu9wms100233b6k66xj1ckr%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Fapp%252Fjsonplaceholders%252F_utils%252Ffetchers.ts%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522DIFF%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Fapp%252Fjsonplaceholders%252F%2540todos%252Fpage.tsx%2522%252C%2522base%2522%253A%2522HEAD%2522%252C%2522id%2522%253A%2522clyudsx0w01793b6krbspgu9t%2522%252C%2522mode%2522%253A%2522temporary%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clyudvx8x00023b6k5b6aafqd%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A4%252C%2522startColumn%2522%253A1%252C%2522endLineNumber%2522%253A4%252C%2522endColumn%2522%253A1%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252Fapp%252Fjsonplaceholders%252F%2540todos%252Fpage.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%257D%252C%2522clnhcfi9h00pi3b6l1d9skltu%2522%253A%257B%2522id%2522%253A%2522clnhcfi9h00pi3b6l1d9skltu%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A3000%252C%2522id%2522%253A%2522clyuaf8dd008k3b6kshozqbmw%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252Fjsonplaceholders%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clyuaf8dd008k3b6kshozqbmw%2522%257D%252C%2522clnhcfi9h00pg3b6lo67cmte3%2522%253A%257B%2522id%2522%253A%2522clnhcfi9h00pg3b6lo67cmte3%2522%252C%2522activeTabId%2522%253A%2522clyudrtk6014j3b6k94z7y94g%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clyua16pr00ch3b6k9rgmauvx%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522clyua16sr002bdmep3h1xg4tb%2522%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522id%2522%253A%2522clyuaf30f006q3b6klilu9qzu%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%252C%257B%2522type%2522%253A%2522GIT_LOG%2522%252C%2522shellId%2522%253A%2522clyudss5q000hdqhrfpz6akby%2522%252C%2522id%2522%253A%2522clyudrtk6014j3b6k94z7y94g%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

**Reply by Abukodonosor** [reply]
Seems that loading.tsx component is needed in the root of the sub-route.

Thanks for feedback.

**Reply by tresorama** [reply]
```tsx
<ErrorBoundary error={ErrorTSX}>
  <Suspense fallback={LoadingTSX}>
    <PageTSX/>
  </Suspense>
</ErrorBoundary>
```
Yes, If I remember well this is the tree 😎 


Edit:
2025-05-15  
This tree is rendered by next. You don't need to add this jsx to your code, but you need to ensure that page, error and loading files are present in your route.

**Reply by Abukodonosor** [reply]
Nop… not the component tree!

**Comment by tresorama** [comment]
All seems working as expected.
If someone found that this problem is still present, ping this discussion and I reopen it. 

For now I close it.

**Reply by Abukodonosor** [reply]
The adding just the file based component!
( Dosenot work )

only when you put it inside boundery and suspense component together!

**Reply by tresorama** [reply]
> The adding just the file based component! ( Dosenot work )
> 
> only when you put it inside boundery and suspense component together!

could you rephrase it ? english is not my first language and I'm not getting the point here...
what is not working ?

**Reply by Abukodonosor** [reply]
No problem.

the NextJS have the unique feature which enable us to use file-based router architecture.
Also there are some protected keywords which make this feature alive like:

layout, loading, error, not-found…


In this context, the loading component dosent work as expected, utill we nake propper workaround.


Hope it helps!

**Reply by tresorama** [reply]
What is not working as expected for the `loading` component?

**Comment by jcmnavia** [comment]
Still happening on nextjs 14.1.3
I don't want to wrap all my routes on Supense components, fix??

**Reply by tresorama** [reply]
Can you share some code or description ?
What is not working?  
Which file tree do you have?
